### PR TITLE
Detect and inform to user if the website is down 

### LINKED ISSUE #66: OpenHub site error is detected as taken
https://www.openhub.net/accounts/45345345 currently responds with "The Black Duck Open Hub is currently offline"

This actually happens quite a bit, so we should have a way to detect and inform the user that OpenHub cant be checked at the moment.

### PR DESCRIPTION
**Pull request by seeeturtle**
This will detect and inform to user if website is down.
It detects the status by checking if constant username
which is probably taken forever is available.

Closes #66 

and some more commits

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,3 +5,4 @@ requests~=2.12
 pyyaml==3.12
 beautifulsoup4==4.6.0
 python-dotenv==0.7.1
+click==6.6 # Temporary fix for coala error with dependecies
```
Comment (nalinbhardwaj) [inline requirements.txt]: dependencies?

Comment (seeeturtle) [reply requirements.txt]: oops. I'll fix it right now.

File: requirements.txt [hunk 2]
```diff
@@ -5,3 +5,4 @@ requests~=2.12
 pyyaml==3.12
 beautifulsoup4==4.6.0
 python-dotenv==0.7.1
+click==6.6 # Temporary fix for coala error with dependencies
```
Comment (manu-chroma) [inline requirements.txt]: move it to test-dep

Comment (seeeturtle) [reply requirements.txt]: hmm... yeah, it's related with test but I thought it should be near with the related library.

Comment (manu-chroma) [reply requirements.txt]: ``coala`` is the related lib here. I think it would be better at the top of the test-dep

**Comment by manu-chroma** [comment]
The changes look good to me.

@jayvdb can also review once, after you get the travis passing :+1:

**Comment by seeeturtle** [comment]
@manu-chroma , it pass now. I think there was some problem with travis

File: tests/test_data.yml [hunk 3]
```diff
@@ -1,4 +1,13 @@
 ---
+asciinema:
+  invalid_usernames:
```
Comment (manu-chroma) [inline tests/test_data.yml]: thanks for this @seeeturtle

File: requirements.txt (hunk 2 above)
Comment (seeeturtle) [reply requirements.txt]: got it.

**Comment by jayvdb** [comment]
I am a little worried that regularly running the same check on a static username is going to cause websites to suspect that this service is a bot and block it.

We could possibly reduce the number of times the check is run.

e.g. only run the check if the user supplied username is available.

e.g. cache the website status for a few minutes, which means the 'usable' check isnt run multiple times when a user trying a few different usernames in quick succession .

**Comment by manu-chroma** [comment]
``caching`` is a great idea 👍

**Comment by seeeturtle** [comment]
@jayvdb, is there a big change with the request number? I don't think it'll block because of requesting one or two more request per a username.

File: username_api.py [hunk 4]
```diff
@@ -118,9 +144,33 @@ def check_username(website, username):
             'url': url,
             'avatar': get_avatar(website, username),
             'possible': possible,
+            'usable': usable,
         }
 
 
+def check_usable(website):
```
Comment (manu-chroma) [inline username_api.py]: just one pick: can you describe the need for this function. just expand a little in the docstring

File: username_api.py [hunk 5]
```diff
@@ -118,9 +144,35 @@ def check_username(website, username):
...
+    """
+        Check if the website is usable.
+        Without checking the website is usable,
```
Comment (manu-chroma) [inline username_api.py]: I don't think this one is very helpful. 

i meant something like, websites go into maintenance unexpectedly. this function checks for verifies if they're working by checking for "known taken" usernames. the checking is rate limited by caching. 

For someone who plans to contribute in the future, i think such a doc string will be helpful to let the know the purpose. pls improve this

Comment (seeeturtle) [reply username_api.py]: got it.

**Pull request merged**